internal/auth/auth.go

```diff
@@ -344,3 +352,68 @@ func persistConfiguration() error {
 	_ = os.MkdirAll(filepath.Dir(path), os.ModePerm)
 	return os.WriteFile(path, data, 0o600)
 }
+
+func RefreshCloudSession() error {
+	if globalConfig == nil {
+		c := DefaultConfig()
+		globalConfig = &c
+	}
+
+	data := url.Values{}
+	data.Set("refresh_token", globalConfig.CloudRefreshToken)
+	data.Set("client_id", cloudIdentityServiceClientId)
+	data.Set("grant_type", "refresh_token")
+
+	req, err := http.NewRequest("POST", cloudIdentityServiceTokenUrl, strings.NewReader(data.Encode()))
+	if err != nil {
+		return fmt.Errorf("failed to create request: %w", err)
+	}
+	req.Header.Set("Content-Type", "application/x-www-form-urlencoded")
+
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return fmt.Errorf("failed to send post request: %w", err)
+	}
+	defer resp.Body.Close()
+
+	var response *api.AccessToken
+
+	err = json.NewDecoder(resp.Body).Decode(&response)
+	if err != nil {
+		return fmt.Errorf("failed to decode response: %w", err)
+	}
+
+	globalConfig.CloudAccessToken = response.Token
+	globalConfig.CloudRefreshToken = response.RefreshToken
+
+	return persistConfiguration()
+}
+
+func IsAccessTokenExpired() (bool, error) {
+	if globalConfig == nil {
```

Comment:
This is a hidden side effect. This function should not modify global state. Just return true if `globalConfig` is nil. We have `loadConfiguration` set on `init(..)` which should load global config if available. If not, we should check for refresh token anyway or fail.